frontend/common/src/components/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,227 @@
+import React from "react";
+import { FieldError, RegisterOptions, useFormContext } from "react-hook-form";
+import { Combobox as ComboboxPrimitive } from "@headlessui/react";
+
+import { useIntl } from "react-intl";
+import InputUnsaved from "../inputPartials/InputUnsaved/InputUnsaved";
+import { useFieldState, useFieldStateStyles } from "../../helpers/formUtils";
+import { Scalars } from "../../api/generated";
+
+import Actions from "./Actions";
+import Label from "./Label";
+
+import { InputError } from "../inputPartials";
+
+interface Option {
+  value: string;
+  label: React.ReactNode;
+}
+
+export interface ComboboxProps
+  extends Omit<
+    React.HTMLProps<HTMLInputElement>,
+    "capture" | "type" | "label"
+  > {
+  id: string;
+  name: string;
+  label: React.ReactNode;
+  context?: string;
+  rules?: RegisterOptions;
+  hideOptional?: boolean;
+  trackUnsaved?: boolean;
+  options: Option[];
```

Comment:
Hmm, I like that idea assuming I fully understand it. My assumption is that you are suggesting we determine the options from a `graphql` query? Like:

```tsx
  const [{data, fetching, error}] = useSkillQuery({ variables: { searchTerm: query }});

  const options = data.skills.map(skill => ({ value: skill.id, label: getLocalizedName(skill.name, locale)}));
```

Reply:
Basically the same thing we did with the users table being server driven.  The imaginary use case would be a user selection form element but instead of loading possibly thousands of user names into the browser it would load Aaron - Allison.  Then I would type "P" and it would load Pamela - Polly.  Then "Pe" would load Pearl - Pythagoras.  We don't have any forms that do this right now (I think?) but it's a fairly common pattern that I hope this component would at least be possible to use with.

Reply:
Yeah, okay that makes perfect sense and I fully agree. I'll see if I can get that working in a reasonable amount of time. If I cannot, would you be willing to push this functionality to a new ticket? Since the initial requirement is for this to be used in #5053 which only loads 184 records.

Reply:
The new story for this is awesome!